Src/ST-LIB_LOW/Sensors/EncoderSensor/EncoderSensor.cpp

```diff
@@ -0,0 +1,85 @@
+#include "ST-LIB_LOW/Sensors/EncoderSensor/EncoderSensor.hpp"
+#include "Encoder/Encoder.hpp"
+#include "Time/Time.hpp"
+
+double d = 0.0;
+
+EncoderSensor::EncoderSensor(Pin pin1, Pin pin2, double *position, double *speed, double *acceleration)
+: position(position), speed(speed), acceleration(acceleration){
+	optional<uint8_t> identification = Encoder::register_encoder(pin1,pin2);
+	if(identification){
+		id = identification.value();
+	}else{
+		//TODO: add Error handler for register here (register returns empty optional)
+		id = 69;
+	}
+}
+
+void EncoderSensor::start(){
+	Encoder::reset_encoder(id);
+	Encoder::turn_on_encoder(id);
+	uint64_t clock_time = Time::get_global_tick();
+	for(int i = 0; i < N_FRAMES; i++){
+		positions[i] = 0.0;
+		times[i] = i*FRAME_SIZE - N_FRAMES*FRAME_SIZE + (((int) clock_time)/ NANO_SECOND);
+		speeds[i] = 0.0;
+	}
+	time = 0.0;
+	last_clock_time = clock_time;
+}
+
+void EncoderSensor::read(){
+	optional<uint32_t> optional_counter = Encoder::get_encoder_counter(id);
+	optional<bool> optional_direction = Encoder::get_encoder_direction(id);
+	uint64_t clock_time = Time::get_global_tick();
+	
+	if(optional_counter && optional_direction){
+		long int delta_clock = clock_time - last_clock_time;
+		if(clock_time < last_clock_time){delta_clock = clock_time + CLOCK_MAX_VALUE * NANO_SECOND / CLOCK_FREQUENCY - last_clock_time;}
```

Comment:
```suggestion
		if(clock_time < last_clock_time) { 
		delta_clock = clock_time + CLOCK_MAX_VALUE * NANO_SECOND / CLOCK_FREQUENCY - last_clock_time;
		}
```
Creo que estas handleando overflow verdad? explicalo poniendole nombre a las operaciones una por una, o pon algun comentario explicativo